src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Browser.cs

```diff
@@ -61,7 +62,8 @@ public async Task ConnectAsync(Uri uri, CancellationToken cancellationToken, Cli
 
                 Abort();
 
-                if (exc is WebSocketException)
+                if (exc is WebSocketException ||
+                    (exc is OperationCanceledException && cancellationToken.IsCancellationRequested))
```

Comment:
I'm not sure what the optimal/expected behavior here is but this check should probably match the BrowserWebSocket case

Reply:
This should already be resolved via commits below by yourself